app/services/remove_status_service.rb

```diff
@@ -57,14 +53,37 @@ def remove_reblogs
     end
   end
 
-  def unpush(type, receiver, status)
-    if status.reblog? && !redis.zscore(FeedManager.instance.key(type, receiver.id), status.reblog_of_id).nil?
-      redis.zadd(FeedManager.instance.key(type, receiver.id), status.reblog_of_id, status.reblog_of_id)
+  def unpush_bulk(type, receiver_ids, status)
```

Comment:
In the case `unpush_bulk_reblog` must always between `if status.reblog?` and `else` and `unpush_bulk_status` must always between `else` and `end`. I do not think such division makes sense.